iceoryx_binding_c/test/test.hpp

```diff
@@ -20,4 +20,19 @@
 #include <gmock/gmock.h>
 #include <gtest/gtest.h>
 
+#include <numeric>
+
+namespace iox_test_binding_c
+{
+template <typename T>
+T maxUnderlyingCEnumValue()
+{
+// ignore the warning since we would like to test the behavior of an invalid enum value
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wconversion"
+    return static_cast<T>(std::numeric_limits<std::underlying_type_t<T>>::max());
```

Comment:
Are you sure this will trigger a warning? The cast should be fine?

Reply:
It triggers the warning from the c2cpp conversion